### ISSUE TITLE
I have nan rpn bbox loss

### ISSUE THREAD

**Issue by dhkwon1122** [matterport/Mask_RCNN#159]
Hi, @waleedka 

I'm trying to fine-tune imagenet pre-trained model to coco task(detection and segmentation) with your code.
I already tried sample train, and to get more precise model, I start with my own pre-trained model(exactly resnet-101).
But I got nan at rpn bbox loss. the other loss seems not that bad,

what can be suspect of this phenomenon?
anything is okay, please give me advice.

Thank you.

**Comment by Liron-2** [comment]
I have this problem too and this error before: 

model.py:1393: RuntimeWarning: divide by zero encountered in log
  np.log(gt_h / a_h)

**Comment by hanskrupakar** [comment]
I have the same problem as well as divide by zero in log.

```
  3/3000 [..............................] - ETA: 149:10:52 - loss: nan - rpn_class_loss: nan - rpn_bbox_loss: nan - mrcnn_class_loss: nan - mrcnn_bbox_loss: 0.5063 - mrcnn_mask_loss: 0.2470
   4/3000 [..............................] - ETA: 156:12:27 - loss: nan - rpn_class_loss: nan - rpn_bbox_loss: nan - mrcnn_class_loss: nan - mrcnn_bbox_loss: 0.3798 - mrcnn_mask_loss: 0.1853
   5/3000 [..............................] - ETA: 190:53:22 - loss: nan - rpn_class_loss: nan - rpn_bbox_loss: nan - mrcnn_class_loss: nan - mrcnn_bbox_loss: 0.3038 - mrcnn_mask_loss: 0.1482
```

**Comment by heidi-z400** [comment]
I was getting a loss of nan. I had 'NUM_CLASSES' set to the number of classes in my training data. 
I added +1 to that for 'BG' (background) and it fixed the problem.

**Comment by viig99** [comment]
I am also getting the same issue, with another dataset
      "Configurations:\n",
      "BACKBONE_SHAPES                [[64 64]\n",
      " [32 32]\n",
      " [16 16]\n",
      " [ 8  8]\n",
      " [ 4  4]]\n",
      "BACKBONE_STRIDES               [4, 8, 16, 32, 64]\n",
      "BATCH_SIZE                     8\n",
      "BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]\n",
      "DETECTION_MAX_INSTANCES        100\n",
      "DETECTION_MIN_CONFIDENCE       0.7\n",
      "DETECTION_NMS_THRESHOLD        0.3\n",
      "GPU_COUNT                      1\n",
      "IMAGES_PER_GPU                 8\n",
      "IMAGE_MAX_DIM                  256\n",
      "IMAGE_MIN_DIM                  256\n",
      "IMAGE_PADDING                  True\n",
      "IMAGE_SHAPE                    [256 256   3]\n",
      "LEARNING_MOMENTUM              0.9\n",
      "LEARNING_RATE                  0.001\n",
      "MASK_POOL_SIZE                 14\n",
      "MASK_SHAPE                     [28, 28]\n",
      "MAX_GT_INSTANCES               100\n",
      "MEAN_PIXEL                     [123.7 116.8 103.9]\n",
      "MINI_MASK_SHAPE                (56, 56)\n",
      "NAME                           databowl\n",
      "NUM_CLASSES                    2\n",
      "POOL_SIZE                      7\n",
      "POST_NMS_ROIS_INFERENCE        1000\n",
      "POST_NMS_ROIS_TRAINING         2000\n",
      "ROI_POSITIVE_RATIO             0.33\n",
      "RPN_ANCHOR_RATIOS              [0.5, 1, 2]\n",
      "RPN_ANCHOR_SCALES              (16, 32, 64, 128, 256)\n",
      "RPN_ANCHOR_STRIDE              1\n",
      "RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]\n",
      "RPN_NMS_THRESHOLD              0.7\n",
      "RPN_TRAIN_ANCHORS_PER_IMAGE    256\n",
      "STEPS_PER_EPOCH                610\n",
      "TRAIN_ROIS_PER_IMAGE           100\n",
      "USE_MINI_MASK                  False\n",
      "USE_RPN_ROIS                   True\n",
      "VALIDATION_STEPS               60\n",
      "WEIGHT_DECAY                   0.0001\n",

**Comment by maksimovkonstantin** [comment]
@waleedka can you help to understand why this problem happens on custom dataset when training from scratch without pretrained coco model?

**Comment by waleedka** [comment]
It's hard to tell without really digging into it and tracking each particular case. A NaN can happen if we divide by zero, or if a number gets too big, so you'd need to watch out for these two cases. One situation that might lead to it is if you start the training using a big learning curve. Try a bit of warm up first, with a very small learning rate (for example, divide the standard learning rate by 100) and after the weights tablize, then increase the learning weight to the normal one. I hope that helps.

**Comment by lianyi** [comment]
I had the same issue. In my case, the cropping will cause boxes to be zero thus the `np.log(gt_h / a_h)` division zero problem. Above PR fixed this specific case.

**Comment by waleedka** [comment]
@lianyi Thank you for the fix! It's merged in the master branch. Closing this issue.

**Comment by mzk912778163** [comment]
> I have this problem too and this error before:
> 
> model.py:1393: RuntimeWarning: divide by zero encountered in log
> np.log(gt_h / a_h)

thks  , after i change it to  np.log(gt_h / (a_h+0.001))  ,it works well.
